File: server/src/main/java/com/smartexpense/api/application/dto/request/ExpenseRequestDto.java
package com.smartexpense.api.application.dto.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
public class ExpenseRequestDto {
    @NotNull(message = "Amount is required")
    private BigDecimal amount;

    @NotBlank(message = "Category is required")
    private String category;

    @NotBlank(message = "Photo URL is required")
    private String photoUrl;

    private String caption;

    @NotNull(message = "Expense date is required")
    private LocalDateTime expenseDate;
}

File: server/src/main/kotlin/com/smartexpense/api/application/dto/request/ExpenseRequestDto.kt
package com.smartexpense.api.application.dto.request

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import java.math.BigDecimal
import java.time.LocalDateTime

data class ExpenseRequestDto(
    @field:NotNull(message = "Amount is required")
    val amount: BigDecimal,

    @field:NotBlank(message = "Category is required")
    val category: String,

    @field:NotBlank(message = "Photo URL is required")
    val photoUrl: String,

    val caption: String? = null,

    @field:NotNull(message = "Expense date is required")
    val expenseDate: LocalDateTime
)
